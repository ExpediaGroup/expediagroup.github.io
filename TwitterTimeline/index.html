<!doctype html>
<html>
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width,initial-scale=1">
<meta name="generator" content="Docusaurus v2.0.0-alpha.72">
<title data-react-helmet="true"></title><link rel="stylesheet" href="/open-source-portal/assets/css/styles.6b81c8d3.css">
<link rel="preload" href="/open-source-portal/assets/js/styles.51523802.js" as="script">
<link rel="preload" href="/open-source-portal/assets/js/runtime~main.8fd641c6.js" as="script">
<link rel="preload" href="/open-source-portal/assets/js/main.e7990a58.js" as="script">
<link rel="preload" href="/open-source-portal/assets/js/c4f5d8e4.f9b81f8d.js" as="script">
<link rel="preload" href="/open-source-portal/assets/js/5.7a7e7710.js" as="script">
<link rel="preload" href="/open-source-portal/assets/js/common.d5adf33b.js" as="script">
</head>
<body>
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=function(){var t=null;try{t=localStorage.getItem("theme")}catch(t){}return t}();t(null!==e?e:"light")}()</script><div id="__docusaurus">
<div><a class="twitter-timeline" data-height="535">Tweets by ExpediaGroupEng</a></div></div>
<script src="/open-source-portal/assets/js/styles.51523802.js"></script>
<script src="/open-source-portal/assets/js/runtime~main.8fd641c6.js"></script>
<script src="/open-source-portal/assets/js/main.e7990a58.js"></script>
<script src="/open-source-portal/assets/js/c4f5d8e4.f9b81f8d.js"></script>
<script src="/open-source-portal/assets/js/5.7a7e7710.js"></script>
<script src="/open-source-portal/assets/js/common.d5adf33b.js"></script>
</body>
</html>