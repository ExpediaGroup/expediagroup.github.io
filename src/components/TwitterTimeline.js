import React, { Component } from 'react';

class TwitterTimeline extends Component {
    constructor(props) {
        super(props);
    }

    componentDidMount() {
        const script = document.createElement("script");

        script.src = "https://platform.twitter.com/widgets.js";
        script.async = true;

        document.body.appendChild(script);
    }

    render(){
        return (
            <div>
                <a
                    className="twitter-timeline"
                    href={this.props.twitterLink}
                    data-height="535">
                    Tweets by ExpediaGroupEng
                </a>
            </div>
        )
    }
}

export default TwitterTimeline;
