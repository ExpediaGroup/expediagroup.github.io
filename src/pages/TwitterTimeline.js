import React, { Component } from 'react';

class TwitterTimeline extends Component {
    constructor(props) {
        super(props);
        this.state = {
            posts: [],
        };
    }

    componentDidMount() {
        const script = document.createElement("script");

        script.src = "https://platform.twitter.com/widgets.js";
        script.async = true;

        document.body.appendChild(script);
    }

    render(){
        return (
            <div>
                <a
                    class="twitter-timeline"
                    href="https://twitter.com/ExpediaGroupEng?ref_src=twsrc%5Etfw"
                    width="320"
                    data-height="480">
                    Tweets by ExpediaGroupEng
                </a>
            </div>
        )
    }
}

export default TwitterTimeline;
