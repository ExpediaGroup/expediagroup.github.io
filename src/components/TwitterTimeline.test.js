/**
 * @jest-environment jsdom
 */

import React from 'react';
import renderer from 'react-test-renderer';
import TwitterTimeline from './TwitterTimeline';

const TWITTER_ACCOUNT_LINK = 'http://test-twitter-account';
const TWITTER_WIDGET_LINK = 'https://platform.twitter.com/widgets.js';

it('renders a skelethon anchor tag that will be hydrated by Twitter script', () => {
    let component;
    renderer.act(() => {
        component = renderer.create(<TwitterTimeline twitterLink={TWITTER_ACCOUNT_LINK}/>);
    });
    expect(component.toJSON()).toMatchSnapshot();
});

it('adds Twitter script to the body', () => {
    jest.spyOn(document.body, 'appendChild');

    renderer.act(() => {
        renderer.create(<TwitterTimeline twitterLink={TWITTER_ACCOUNT_LINK}/>);
    });

    const appendedScript = document.body.appendChild.mock.calls[0][0];
    expect(appendedScript).toBeInstanceOf(HTMLScriptElement);
    expect(appendedScript).toHaveProperty("src", TWITTER_WIDGET_LINK);
});
