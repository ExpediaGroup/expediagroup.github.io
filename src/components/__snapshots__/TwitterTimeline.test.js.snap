// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a skelethon anchor tag that will be hydrated by Twitter script 1`] = `
<div>
  <a
    className="twitter-timeline"
    data-chrome="noheader nofooter"
    data-height="525"
    href="http://test-twitter-account"
  >
    <h3>
      Loading tweets..
    </h3>
  </a>
</div>
`;
